### ISSUE TITLE
Preset colors from theme.json aren't loaded in the editor

### ISSUE THREAD

**Issue by markhowellsmead** [WordPress/gutenberg#33822]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Have you tried deactivating all plugins except Gutenberg?

- [X] I have tested with all plugins deactivated.

### Have you tried replicating the bug using a default theme e.g. Twenty Twenty?

- [X] I have tested with a default theme.

### Description

I’ve added theme.json to my Theme and added some colors to the palette array. e.g. `--wp--preset--color--green`. Then I’ve specified that links without any specific `.has-text-color` class should use this color as a default: `a:not(.has-text-color) {color: var(--wp--preset--color--green );}` I’ve added this to my CSS files enqueued in the frontend and in the editor.

This works fine in the frontend, because WP loads the custom properties in the head of the page. But the editor doesn’t have the `--wp--preset--color--green` custom property loaded in the editor, so the links are all black.

### Step-by-step reproduction instructions

1. Add colors to the palette array in theme.json.
2. Add rules to the enqueued editor CSS file which use the custom properties.
3. The custom properties are not defined in the editor.

### Expected Behavior

The editor should provide the values/custom properties in the editor.

### Current Behavior

The editor does not provide the values/custom properties in the editor.

### WordPress Information

Version | 5.8
Sprache der Website | de_DE_formal
Benutzersprache | de_DE_formal
Zeitzone | Europe/Zurich
Homepage-URL | https://ffde.local
Website-URL | https://ffde.local
Permalink-Struktur | /%postname%/
Verwendet diese Website HTTPS? | Ja
Ist dies eine Multisite? | Nein
Kann sich jeder auf dieser Website registrieren? | Nein
Ist diese Website für Suchmaschinen ausgeschlossen? | Ja
Standard-Kommentarstatus | Offen
Umgebungstyp | local
Anzahl der Benutzer | 1
Verbindung mit WordPress.org | WordPress.org ist erreichbar

### Gutenberg Information

Not installed.

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Microsoft Edge, Other

### Device Information

Desktop and iPhone

### Operating System Information

Mac, Windows, iOS.

**Comment by jasmussen** [comment]
Let me see if I can help debug this. Just looking at TT1 Blocks, it registers a slew of colors in theme.json like so:

```
	"settings": {
		"color": {
			"link": true,
			"palette": [
				{
					"slug": "black",
					"color": "#000000",
					"name": "Black"
				},
...
```

With just those I can use the color swatches from the editor to colorize a paragraph, those are tied to CSS classes and output even without loading any styles or editor styles (e.g. `has-blue-color has-dark-gray-background-color`):

<img width="456" alt="Screenshot 2021-08-05 at 14 58 17" src="https://user-images.githubusercontent.com/1204802/128354069-617982c3-70f5-488a-a369-ebc7142d1f9f.png">

I understand you load an editor style (add_editor_style) that then references the colors defined in your theme.json for custom rules. In testing just now, adding a color called "test", I did see it registered as `--wp--preset--color--test`, so it should be output for your usage. Can you open the web-inspector and see if that property is even output on the frontend and in the editor?

<img width="587" alt="Screenshot 2021-08-05 at 15 05 58" src="https://user-images.githubusercontent.com/1204802/128354839-069ce5d9-4772-41f0-9edc-dcae34906e91.png">

If they are, can you try a simpler rule than `a:not(.has-text-color) {color: var(--wp--preset--color--green );}` and see if that helps? 

That rule on its own wouldn't work in my theme, by the way, as I don't have any blocks that would apply the `has-text-color` on the `a` element directly. Manually coloring a link gives me this markup:

```
<a href="http://example.com"><span class="has-inline-color has-red-color">example link</span></a>
```

Could it be a typo in the rule?

**Comment by markhowellsmead** [comment]
Using the browser inspector, I can see the same result as in your screenshots. The properties are output in the frontend. However, again using the inspector, I can see that the properties are not output in the editor.

The issue is not with the CSS (e.g. for the links), as they work fine if I use a fixed hexadecimal value, but in the fact that the properties aren't defined in the editor.

**Comment by jasmussen** [comment]
That's curious. Are you able to provide me with a copy of your theme.json file? You can trim out any private bits if need be, I'll try it on my end. Thank you.

**Comment by markhowellsmead** [comment]
```
{
    "version": 1,
    "settings": {
        "layout": {
            "contentSize": "736px",
            "wideSize": "1124px"
        },
        "spacing": {
            "customMargin": false,
            "customPadding": false
        },
        "color": {
            "palette": [
                {
                    "slug": "green",
                    "color": "#73ce9a",
                    "name": "Green"
                },
                {
                    "slug": "black",
                    "color": "#000000",
                    "name": "Black"
                },
                {
                    "slug": "white",
                    "color": "#FFFFFF",
                    "name": "White"
                },
                {
                    "slug": "dark-grey",
                    "color": "#121212",
                    "name": "Dark grey"
                },
                {
                    "slug": "medium-grey",
                    "color": "#383838",
                    "name": "Medium grey"
                }
            ]
        }
    }
}

```

**Comment by jasmussen** [comment]
I see them on my end.

<img width="1811" alt="Screenshot 2021-08-05 at 15 19 53" src="https://user-images.githubusercontent.com/1204802/128356756-844a7eeb-d270-4291-a7cb-899894e46029.png">

When I disable the Gutenberg plugin, I can reproduce, and I do not see the custom properties output:

<img width="1811" alt="Screenshot 2021-08-05 at 15 23 57" src="https://user-images.githubusercontent.com/1204802/128357340-13bd0881-9b72-4de7-a96d-66dbd0a95e5f.png">

@nosolosw @jorgefilipecosta I know there are a couple of full site editing features not shipping in 5.8, and requiring the plugin in order to work. Is the palette part of theme.json meant to work?

**Comment by oandregal** [comment]
Nah, the palette should work without the Gutenberg plugin.

I've just tested with WordPress 5.8 with the Gutenberg plugin deactivated. I've used the empty theme and substituted its `theme.json` by [the one shared above](https://github.com/WordPress/gutenberg/issues/33822#issuecomment-893451384). It's working fine for me:

![Captura de ecrã de 2021-08-05 17-03-21](https://user-images.githubusercontent.com/583546/128373762-b3893dd4-7c36-4922-94c7-dce2104ce75d.png)

I also tested with the latest Gutenberg active (11.2.1) and works as expected. I can only think of two reasons why it won't work without the plugin: the WordPress version is not WordPress 5.8 or the file is called `experimental-theme.json` instead of `theme.json` (we still have a fallback in the plugin for the first name, but that wasn't merged in WordPress 5.8). :thinking:

**Comment by markhowellsmead** [comment]
Thanks for the testing. The file is theme.json and is in the root folder of the Theme. WordPress is at 5.8.

**Comment by jasmussen** [comment]
In my case, my file _was_ named `experimental-theme.json`, and renaming it to `theme.json` fixed it for me. So now I can't reproduce. I'm curious what might be the issue — just for debugging purposes, does it work if you install and activate the Gutenberg plugin?

**Comment by markhowellsmead** [comment]
Crap. Found it. 🤦 We based our development on our own starter Theme. Because WordPress used to output custom properties in the header which weren't used anywhere in the code, we deactivated the "styles" part of the block settings, via the `block_editor_settings_all` hook. Removing that old code allows the styles block (and hence the new custom properties) to be added to the editor as expected.
